segment/types.go

```diff
@@ -27,6 +28,44 @@ type Source struct {
 	CreateTime    time.Time     `json:"create_time,omitempty"`
 }
 
+// CommonEventSettings provides accepted values for CommonTrackEventOnViolations, CommonIdentifyEventOnViolations and CommonGroupEventOnViolations
+type CommonEventSettings string
+
+const (
+	Allow CommonEventSettings = "ALLOW"
+
+	// Only for use with CommonTrackEventOnViolations
+	OmitProps = "OMIT_PROPERTIES"
+
+	// Only for use with CommonIdentifyEventOnViolations and CommonIdentifyEventOnViolations
+	OmitTraits = "OMIT_TRAITS"
+
+	Block = "BLOCK"
```

Comment:
should these all be of the same type, i.e. `CommonEventSettings` ?

Reply:
It's strange, I thought that the first type was getting propagated to the other declarations (I got it from https://www.ribice.ba/golang-enums/). But vscode does see them as string 🤔  I'll put it on each line it'll be clearer